html/cross-origin-opener-policy/header-parsing.https.html

```diff
@@ -0,0 +1,52 @@
+<!doctype html>
+<meta charset=utf-8>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="resources/common.js"></script>
+
+<div id=log></div>
+<script>
+
+let tests = [
+  // popup Origin, popup COOP, expect opener
+
+  // None of the following should be recognized as "same-origin" (hence the
+  // "expected opener" value of `true`).
+  [SAME_ORIGIN, "same\u2014origin", true], // non-ASCII character (em dash)
```

Comment:
Having seen the COEP comments I guess we should address here too what bytes are going to be emitted by the server. And what shortcomings this approach has (in that 0xFF isn't feasible). Perhaps we should change approach so we can effectively specify bytes?

Reply:
Yeah, good point. I've changed it now for this PR to specify bytes.

Reply:
Did you check that other users of common.js not rely on encodeURIComponent? Especially reporting tests might be impacted, perhaps?

Reply:
Hmm yes, though it's not obvious from test results ([chrome](https://wpt.fyi/results/html/cross-origin-opener-policy?diff&filter=ADC&run_id=670010001&run_id=668060003), [firefox](https://wpt.fyi/results/html/cross-origin-opener-policy?diff&filter=ADC&run_id=648910001&run_id=648890005)).

But instead of sprinkling `encodeURIComponent` everywhere, I'm thinking about doing it differently, like using an object for the 0xFF test case:

```
  [SAME_ORIGIN, { percentEscaped: "same%FForigin" }, true]
```

Reply:
That seems reasonable, though I'd name it `percentEncoded`.

Reply:
Done.